src/common/components/Button/index.types.ts

```diff
@@ -0,0 +1,6 @@
+import { ReactNode } from '@tanstack/react-router';
+import { ComponentPropsWithoutRef } from 'react';
+
+export interface ButtonPropsType extends ComponentPropsWithoutRef<'button'> {
```

Comment:
```suggestion
export interface ButtonPropsType extends React.ComponentPropsWithoutRef<'button'> {
```

이렇게 React.으로 참조하면 import 안해와도 되더라구요

Reply:
namespace를 명시하는 게 더 직관적일까요?
예제에서도 type 같은 경우에는 namespace를 명시하는 경우가 잦더라고요.

Reply:
일단 namespace 포함시키는 방향으로 수정해볼게요

Reply:
만약 react namespace를 다른 곳에서도 많이 쓴다면 import 해오는 것이 더 직관적이겠으나, 이번 같은 경우는 한 줄에서만 써서 namespace를 명시하는 것이 더 간편해 보이는 것 같아요!

Reply:
resolved [633e6ae](https://github.com/uxplorerss/uxplorers-frontend/pull/29/commits/633e6ae62b50f269faa6caed371d8d3e5191e07f)

Reply:
지나가다가 봤습니다 ㅎㅎㅎ 이 경우에는 `import type { ComponentPropsWithoutRef } from 'react';` 와 같이 import하는 것은 어떤가요? type으로만 쓰고 있다는 것이 명확해지고, type으로 import한 경우에는 빌드 시점에 import가 사라지기도 합니다. 이걸 자동화하는 import eslint룰도 있으니 꽤 많이 쓰이는 컨벤션으로도 보이고요

Reply:
@shinwonse 오 지나가다가 귀한 리뷰 감사합니다
type의 경우 빌드 시점에 사라지는건 처음 알았네요!!

Reply:
@Turtle-Hwan 


reference: https://www.reddit.com/r/typescript/comments/1asfe20/comment/kqwz50l/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

> 지나가다가 봤습니다 ㅎㅎㅎ 이 경우에는 `import type { ComponentPropsWithoutRef } from 'react';` 와 같이 import하는 것은 어떤가요? type으로만 쓰고 있다는 것이 명확해지고, type으로 import한 경우에는 빌드 시점에 import가 사라지기도 합니다. 이걸 자동화하는 import eslint룰도 있으니 꽤 많이 쓰이는 컨벤션으로도 보이고요

@shinwonse 
피드백 감사합니다.
import type을 쓰는 이유를 다음과 같이 말씀해주셨는데요.
1. self-explanatory(타입을 구분 가능)
2. 빌드 시점에 import가 사라진다

verbatimModuleSyntax flag가 체크되어있지 않다면 (2)의 이점은 없는 거 같습니다. 기본적으로 type만 있는 import의 경우 compiled output에서는 지워진다네요.

<img width="737" alt="image" src="https://github.com/user-attachments/assets/c71f6580-367e-4f93-b193-7ebd6600c481">
https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax

또한 verbatimModuleSyntax flag를 체크하면, type만 있는 import더라도 `import type`을 명시하지 않으면 module import가 되기 때문에 굳이 할 필요를 느끼지는 못하겠네요.

<img width="447" alt="image" src="https://github.com/user-attachments/assets/b76ddd4c-d63c-4180-a473-7d8eaa9d141e">

따라서 (1) readability를 챙겨주기 위해서 `import type`을 사용해보겠습니다! @Turtle-Hwan

Reply:
`verbatimModuleSyntax flag`는 저도 처음 들어보는 개념이네요 감사합니다 ㅎㅎㅎ